tests/test_plugin_karma.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+test_onebot_karma
+----------------------------------
+
+Tests for Karma module.
+"""
+
+from irc3.testing import BotTestCase, patch
+
+from .test_plugin_users import MockDb
+from onebot.plugins.karma import _format_key
+
+
+class KarmaTestCase(BotTestCase):
+
+    config = {
+        'cmd': '!',
+        'includes': [
+            'onebot.plugins.karma',
+        ],
+    }
+
+    @patch('irc3.plugins.storage.Storage')
+    def setUp(self, mock):
+        super(KarmaTestCase, self).setUp()
+        self.callFTU()
+        self.bot.db = MockDb()
+        self.bot.dispatch(':bar!foo@host JOIN #chan')
+
+    def assertSent(self, lines):
+        """Assert that these lines have been sent"""
+        self.assertEqual(self.bot.sent, lines)
+
+    def test_increment(self):
+        self.bot.dispatch(':root@localhost PRIVMSG #chan :foo bar++ baz')
+        self.assertEquals(self.bot.db[_format_key('bar')]['up'], 1)
+        self.assertEquals(self.bot.db[_format_key('bar')]['down'], 0)
+
+    def test_increment_brackets(self):
+        self.bot.dispatch(':root@localhost PRIVMSG #chan :(foo bar)++ baz')
+        self.assertEquals(self.bot.db[_format_key('foo bar')]['up'], 1)
+        self.assertEquals(self.bot.db[_format_key('foo bar')]['down'], 0)
+        with self.assertRaises(KeyError):
+            self.bot.db[_format_key('bar')]
+
+    def test_increment_multiple(self):
+        self.bot.dispatch(':root@localhost PRIVMSG #chan :foo++ foo++ bar++')
+        self.assertEquals(self.bot.db[_format_key('foo')]['up'], 2)
+        self.assertEquals(self.bot.db[_format_key('bar')]['up'], 1)
+
+    def test_decrement_multiple(self):
+        self.bot.dispatch(':root@localhost PRIVMSG #chan :foo++ (foo)++ bar++')
+        self.assertEquals(self.bot.db[_format_key('foo')]['up'], 2)
+        self.assertEquals(self.bot.db[_format_key('bar')]['up'], 1)
+
+    def test_increment_decrement_mix(self):
+        self.bot.dispatch(':root@localhost PRIVMSG #chan :foo++ foo-- bar++')
+        self.assertEquals(self.bot.db[_format_key('foo')]['up'], 1)
+        self.assertEquals(self.bot.db[_format_key('foo')]['down'], 1)
+        self.assertEquals(self.bot.db[_format_key('bar')]['up'], 1)
+
+    def test_decrement(self):
+        self.bot.dispatch(':root@localhost PRIVMSG #chan :foo bar-- baz')
+        self.assertEquals(self.bot.db[_format_key('bar')]['up'], 0)
+        self.assertEquals(self.bot.db[_format_key('bar')]['down'], 1)
+
+    def test_decrement_brackets(self):
+        self.bot.dispatch(':root@localhost PRIVMSG #chan :(foo bar)-- baz')
+        self.assertEquals(self.bot.db[_format_key('foo bar')]['up'], 0)
+        self.assertEquals(self.bot.db[_format_key('foo bar')]['down'], 1)
+        with self.assertRaises(KeyError):
+            self.bot.db[_format_key('bar')]
+
+    def test_chained(self):
```

Comment:
It seems weird that this behaviour is different from the behaviour when doing `(foo--)++`, also it seems confusing that it has two side effects...

It may make the regular expressions overcomplicated though.